packages/commons/src/Utility.ts

```diff
@@ -51,15 +51,46 @@
  * ```
  */
 export class Utility {
+  readonly #initializationType:
+    | 'unknown'
+    | 'on-demand'
+    | 'provisioned-concurrency';
   protected coldStart = true;
   protected readonly defaultServiceName: string = 'service_undefined';
 
+  public constructor() {
+    this.#initializationType = this.getInitializationType();
+    if (this.#initializationType !== 'on-demand') {
+      this.coldStart = false;
+    }
+  }
+
+  /**
+   * Get the value of the `AWS_LAMBDA_INITIALIZATION_TYPE` environment variable.
+   */
+  protected getInitializationType():
+    | 'unknown'
+    | 'on-demand'
+    | 'provisioned-concurrency' {
```

Comment:
NodeJS doesn't support snap-start currently, but we never know if it will in the future. I don't think it hurts to add the `snap-start` value here and it doesn't need to change in the future. Snap-start behaves the same as on-demand when it comes to ColdStart.

But its up to you.